src/traits/fp.rs

```diff
@@ -58,3 +58,39 @@ pub trait FPMatrix {
     where
         F: Fn(Vec<f64>) -> f64;
 }
+
+/// Functional Programming for Complex Matrix
+pub trait FPComplexMatrix {
```

Comment:
Note: In next PRs, this can be removed and the handling added to `pub trait FPMatrix` (with a generic)